internal/dataplane/parser/translate_httproute_test.go

```diff
@@ -849,26 +860,31 @@ func TestIngressRulesFromHTTPRoutes_RegexPrefix(t *testing.T) {
 			},
 		},
 	} {
-		t.Run(tt.msg, func(t *testing.T) {
-			ingressRules := newIngressRules()
+		withParser := func(p *Parser) func(t *testing.T) {
+			return func(t *testing.T) {
+				ingressRules := newIngressRules()
 
-			var errs []error
-			for _, httproute := range tt.routes {
-				// initialize the HTTPRoute object
-				httproute.SetGroupVersionKind(httprouteGVK)
+				var errs []error
+				for _, httproute := range tt.routes {
+					// initialize the HTTPRoute object
+					httproute.SetGroupVersionKind(httprouteGVK)
 
-				// generate the ingress rules
-				err := p.ingressRulesFromHTTPRoute(&ingressRules, httproute)
-				if err != nil {
-					errs = append(errs, err)
+					// generate the ingress rules
+					err := p.ingressRulesFromHTTPRoute(&ingressRules, httproute)
+					if err != nil {
+						errs = append(errs, err)
+					}
 				}
-			}
 
-			// verify that we receive the expected values
-			assert.Equal(t, tt.expected, ingressRules)
+				// verify that we receive the expected values
+				assert.Equal(t, tt.expected, ingressRules)
 
-			// verify that we receive any and all expected errors
-			assert.Equal(t, tt.errs, errs)
-		})
+				// verify that we receive any and all expected errors
+				assert.Equal(t, tt.errs, errs)
+			}
+		}
+
+		t.Run(tt.msg+" using lagacy parser", withParser(parser))
```

Comment:
Should we be able to simply drop in the alternative parser here? The lack of changes to the cases seems to suggest that we weren't testing any cases that would be affected by the gate before. We probably want an independent unit test for `ingressRulesFromHTTPRouteWithCombinedServiceRoutes()` and should then be able to leave this as-is as the de-facto test for the legacy function.

Integration tests will take care of the two variants' functionality after KIC actually applies configuration, though we may want to expand the HTTPRoute test there to include rules that would be affected (two rules that share a backendRef Service and one that doesn't).

Reply:
Sure, but the new test case would be mostly copy paste with some added use cases.

Reply:
@jrsmroz 
> In order not to copy the this huge tests configuration I've moved them to global var.

Please let's not. This can introduce all sorts of issues. I believe your suggestion already leads us in the right way by introducing the function that would return this (huge) struct.

Reply:
You're suggesting a test builder/constructor function?